### PR TITLE
Adding unit tests

### PR DESCRIPTION
**Pull request by gmcnaughton**
Added a handful of unit tests:
* `setObject`: setting the same key twice overwrites the original value
* `objectForKey`: getting a key that does not exist returns nil
* `removeAllObjects`: removes all objects from the cache
* `initWithName`: caches with the same name share the same contents

### CODE REVIEW

File: tests/PINCacheTests/PINCacheTests.m [hunk 1]
```diff
@@ -95,6 +95,28 @@ - (void)testObjectSet
     XCTAssertNotNil(image, @"object was not set");
 }
 
+- (void)testObjectSetWithDuplicateKey
+{
+    NSString *key = @"key";
+    NSString *value1 = @"value1";
+    NSString *value2 = @"value2";
+    __block NSString *value = nil;
```
Comment (garrettmoon) [inline tests/PINCacheTests/PINCacheTests.m]: Mind renaming this to something indicating it's the returned value? Maybe cachedValue?

File: tests/PINCacheTests/PINCacheTests.m [hunk 2]
```diff
@@ -131,6 +172,29 @@ - (void)testObjectRemove
...
+    }];
+    
+    dispatch_semaphore_wait(semaphore, [self timeout]);
+    
+    id object1 = [self.cache objectForKey:key1];
+    id object2 = [self.cache objectForKey:key2];
+    
+    XCTAssertNil(object1, @"not all objects were removed");
+    XCTAssertNil(object2, @"not all objects were removed");
+    XCTAssertTrue(self.cache.memoryCache.totalCost == 0, @"memory cache was not 0 after removing all objects");
```
Comment (garrettmoon) [inline tests/PINCacheTests/PINCacheTests.m]: Mind adding this too:
    XCTAssertTrue(self.cache.diskByteCount == 0, @"disk cache byte count was not 0 after removing all objects");

**Comment by garrettmoon** [comment]
Thank you for writing these! If you don't mind addressing a couple of comments I'll get this merged in!

File: tests/PINCacheTests/PINCacheTests.m (hunk 1 above)
Comment (gmcnaughton) [inline tests/PINCacheTests/PINCacheTests.m]: Yep, renamed to cachedValue!

File: tests/PINCacheTests/PINCacheTests.m (hunk 2 above)
Comment (gmcnaughton) [inline tests/PINCacheTests/PINCacheTests.m]: Yep of course - added this assert!

**Comment by gmcnaughton** [comment]
@garrettmoon Thanks!  I made the changes you suggested and force-pushed the branch.

**Pull request merged**